### PR TITLE
domain gap

### PR DESCRIPTION
**Pull request by yotamazriel**
- init new project
- full domain gap dataset
- packages
- readme v1

### CODE REVIEW

File: examples/domain_gap/README.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+## Tensorleap **Dataset Deployment**
+
+
+To deploy your local changes:
+
+
+```bash
+tensorleap datasets save
+```
```
Comment (yotamazriel) [inline examples/domain_gap/README.md:78]: We need to test it what happens if using save on a clean environment?

File: examples/domain_gap/README.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+.tensorleap.yaml file is configured to a dataset in your Tensorleap environment and is synced to the dataset saved in the environment. 
+
+For any additional file being used we add its path under `include` parameter:
+
+```yaml
+include:
+  - tensorleap.py
+
+```
```
Comment (yotamazriel) [inline examples/domain_gap/README.md:96]: Let's add here the specific files we are uploading in the domain gap example

File: examples/domain_gap/README.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+The **`tool.poetry.scripts`** section is used to map a script name to a Python function or an executable command. When you define a script in this section, it becomes accessible as a command that can be executed from the command line using **`poetry run`**. 
+
+You can add to **`pyproject.toml`**
+
+```python
+[tool.poetry.scripts]
+my-test = "my_package.module:function"
+```
+
+After defining the script, you can execute it using the Poetry CLI:
```
Comment (yotamazriel) [inline examples/domain_gap/README.md:121]: Let's replace this section with a description of how to run the poetry run test command.

File: examples/domain_gap/tests/test_tensorleap.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        for i in range(sub.length):
+            try:
+                x = input_image(i, sub)
+                mask = ground_truth_mask(i, sub)
+                city_val = metadata_city(i, sub)
+                err_i = 0
+            except Exception as e:
+                print(f"Error in processing sample {i} from {sub['subset_name']} set")
+                err_i += 1
+                if err_i > 10:
```
Comment (tomkoren21) [inline examples/domain_gap/tests/test_tensorleap.py:17]: Why is this line here and we stop at 10? is the same thing happens on the engine? o.w. I would not continue the script here - but let the error propogate all the way through (either with print(e) or without the try - except)

File: examples/domain_gap/tests/test_tensorleap.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+if __name__ == "__main__":
+
+    data_subsets: List[PreprocessResponse] = subset_images()  # preprocess and get data preprocess response list
+    i, train = 0, data_subsets[0]
+    img = input_image(i, train)
+    mask = ground_truth_mask(i, train)
+    plot_mask(img, mask)     # plot the GT
+    iter_all_samples(data_subsets)
+
+    MODEL_PATH = None   # TODO: FILL here your path model
```
Comment (tomkoren21) [inline examples/domain_gap/tests/test_tensorleap.py:42]: WDYT on pulling the model from a common GCS folder? I think this might indicate to the user what model should be here (h5) and let him test this flow end-to-end by himself. If we later create a template for users it should be exactly like here IMO - this works well.

**Review by tomkoren21 — COMMENTED** [review]
LGTM. I really like the test script general flow - I think we need to add *all* metadata/loss/visualizers and actually download the model and infer on it. Other than that this works IMO.
